pkg/discovery/worker/controller_metrics_collector.go

```diff
@@ -0,0 +1,171 @@
+		collector.updateQuotaResourcesUsed(kubeController, pod)
+	}
+	return kubeControllerList, nil
+}
+
+// Get KubeController info from the given pod from metrics sink, including controller type, name and UID.
+func (collector *ControllerMetricsCollector) getKubeControllerInfo(entityType metrics.DiscoveredEntityType,
+	entityKey string) (string, string, string, error) {
+	controllerType, err := collector.getOwnerMetric(entityType, entityKey, metrics.OwnerType)
+	if err != nil {
+		return "", "", "", err
+	}
+	controllerName, err := collector.getOwnerMetric(entityType, entityKey, metrics.Owner)
+	if err != nil {
+		return "", "", "", err
+	}
+	controllerUID, err := collector.getOwnerMetric(entityType, entityKey, metrics.OwnerUID)
+	if err != nil {
+		return "", "", "", err
+	}
+	return controllerType, controllerName, controllerUID, nil
+}
+
+// Get Pod owner metric value of a given metric type from metrics sink. Metric type can be Owner (owner name), OwnerType
+// or OwnerUID.
+func (collector *ControllerMetricsCollector) getOwnerMetric(entityType metrics.DiscoveredEntityType, entityKey string,
+	ownerMetricType metrics.ResourceType) (string, error) {
+	ownerMetricId := metrics.GenerateEntityStateMetricUID(entityType, entityKey, ownerMetricType)
+	ownerMetric, err := collector.metricsSink.GetMetric(ownerMetricId)
+	if err != nil {
+		return "", fmt.Errorf("error getting %s from metrics sink for pod %s --> %v", ownerMetricType, entityKey, err)
+	}
+	ownerMetricValue := ownerMetric.GetValue()
+	metricValue, ok := ownerMetricValue.(string)
+	if !ok {
+		return "", fmt.Errorf("error getting %s from metrics sink for pod %s", ownerMetricType, entityKey)
+	}
+	return metricValue, nil
+}
+
+// Update quota resources used for the controller from pod quota usage. The usage values of CPU resources are converted
+// from cores to MHz based on the CPU frequency of the corresponding node where the given pod is located.
+func (collector *ControllerMetricsCollector) updateQuotaResourcesUsed(kubeController *repository.KubeController, pod *api.Pod) {
+	for _, resourceType := range metrics.QuotaResources {
+		podKey := util.PodKeyFunc(pod)
+		metricId := metrics.GenerateEntityResourceMetricUID(metrics.PodType, podKey, resourceType, metrics.Used)
+		metric, err := collector.metricsSink.GetMetric(metricId)
+		if err != nil {
+			glog.Errorf("Error getting %s used value from metrics sink for pod %s: %v", resourceType, podKey, err)
+			continue
+		}
+		resourceUsed := metric.GetValue().(float64)
+		if metrics.IsCPUType(resourceType) {
+			// For CPU resources, convert the usage values expressed in cores to MHz based on the CPU frequency of the
+			// corresponding node where the given pod is located
+			cpuFrequency, err := collector.getNodeCPUFrequency(pod)
+			if err != nil {
+				glog.Errorf("Error getting node cpuFrequency from pod %s: %v", podKey, err)
+				continue
+			}
+			if cpuFrequency > 0.0 {
+				resourceUsed *= cpuFrequency
+			}
```

Comment:
nit: Do we want to log this conversion as well since we log the capacity conversions?